packages/@aws-cdk/region-settings/lib/region.ts

```diff
@@ -0,0 +1,186 @@
+   */
+  public constructor(public readonly name: string, props: RegionProps) {
+    this.partition = props.partition || 'aws';
+    this.domainSuffix = props.domainSuffix || 'amazonaws.com';
+    this.hasCdkMetadataResource = props.hasCdkMetadataResource !== false;
+
+    const edgeHostedZoneId = props.apiGatewayEdgeOptimizedHostedZoneId == null
+      ? 'Z2FDTNDATAQYW2'
+      : props.apiGatewayEdgeOptimizedHostedZoneId;
+    this.apiGateway = {
+      hostedZoneIds: edgeHostedZoneId || props.apiGatewayRegionalHostedZoneId
+        ? { edgeOptimized: edgeHostedZoneId, regional: props.apiGatewayRegionalHostedZoneId }
+        : undefined,
+    };
+
+    this.s3StaticWebsite = {
+      legacyNameFormat: !!props.s3StaticWebsiteLegacyNameFormat,
+      hostedZoneId: props.s3StaticWebsiteHostedZoneId,
+    };
+
+    this.servicePrincipalMaker = props.servicePrincipalMaker || DefaultServicePrincipalMaker.instance;
+  }
+
+  /**
+   * Computes the appropriate service principal for a given service short name.
+   *
+   * @param service the name of the service for which a principal is requested. For example `sns`, `s3`, ...
+   *
+   * @returns a service principal name, such as `s3.amazonaws.com`.
+   */
+  public servicePrincipal(service: string): string {
+    return this.servicePrincipalMaker.makeServicePrincipal(service, this);
+  }
+}
+
+export interface RegionProps {
+  /**
+   * The partition in which a region is (e.g: `aws`, `aws-cn`, ...)
+   *
+   * @default aws
+   */
+  partition?: string;
+
+  /**
+   * The domain suffix for the partition
+   * @default amazonaws.com
+   */
+  domainSuffix?: string;
+
+  /**
+   * Whether the AWS::CDK::Metadata CloudFormation resource is available.
+   *
+   * @default true
+   */
+  hasCdkMetadataResource?: boolean;
+
+  /**
+   * Hosted zone ID for the edge-optimized endpoint
+   * @default Z2FDTNDATAQYW2
+   */
+  apiGatewayEdgeOptimizedHostedZoneId?: string;
+
+  /**
+   * Hosted zone ID for the regional endpoint
+   * @default none
+   */
+  apiGatewayRegionalHostedZoneId?: string;
+
+  /**
+   * Whether the legacy `s3-website-${region}.${domain}` endpoint format is used in that region.
+   * @default false
+   */
+  s3StaticWebsiteLegacyNameFormat?: boolean;
+
+  /**
+   * The Hosted Zone ID for S3 static websites in the region.
+   * @default none
+   */
+  s3StaticWebsiteHostedZoneId?: string;
+
+  /**
+   * The service principal maker for the region.
+   * @default DefaultServicePrincipalMaker
+   */
+  servicePrincipalMaker?: IServicePrincipalMaker;
+}
+
+// Loading all the built-in configurations.
+for (const name of fs.readdirSync(path.join(__dirname, 'settings'))) {
+  if (!name.endsWith('.js')) { continue; }
+  const moduleName = `./settings/${name}`;
+  // tslint:disable-next-line:no-var-requires
+  const module = require(moduleName);
+  let foundRegion = false;
+  for (const exportedValue of Object.values(module)) {
+    if (exportedValue instanceof Region) {
```

Comment:
instanceof! Eek!

Reply:
`peerDependencies` say there should be only one `Region` class in the whole closure.